gax/src/main/java/com/google/api/gax/rpc/RetryingServerStreamingCallable.java

```diff
@@ -0,0 +1,110 @@
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.rpc;
+
+import com.google.api.core.ApiFutureCallback;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.retrying.RetryingFuture;
+import com.google.api.gax.retrying.ScheduledRetryingExecutor;
+import com.google.api.gax.retrying.ServerStreamingAttemptException;
+import com.google.api.gax.retrying.StreamResumptionStrategy;
+
+/**
+ * A ServerStreamingCallable that implements resumable retries.
+ *
+ * <p>Wraps a request, a {@link ResponseObserver} and an inner {@link ServerStreamingCallable} and
+ * coordinates retries between them. When the inner callable throws an error, this class will
+ * schedule retries using the configured {@link ScheduledRetryingExecutor}.
+ *
+ * <p>Streams can be resumed using a {@link StreamResumptionStrategy}. The {@link
+ * StreamResumptionStrategy} is notified of incoming responses and is expected to track the progress
+ * of the stream. Upon receiving an error, the {@link StreamResumptionStrategy} is asked to modify
+ * the original request to resume the stream.
+ *
+ * <p>Package-private for internal use.
+ */
+final class RetryingServerStreamingCallable<RequestT, ResponseT>
+    extends ServerStreamingCallable<RequestT, ResponseT> {
+
+  private final Watchdog<ResponseT> watchdog;
+  private final ServerStreamingCallable<RequestT, ResponseT> innerCallable;
+  private final ScheduledRetryingExecutor<Void> executor;
+  private final StreamResumptionStrategy<RequestT, ResponseT> resumptionStrategyPrototype;
+
+  RetryingServerStreamingCallable(
+      Watchdog<ResponseT> watchdog,
+      ServerStreamingCallable<RequestT, ResponseT> innerCallable,
+      ScheduledRetryingExecutor<Void> executor,
+      StreamResumptionStrategy<RequestT, ResponseT> resumptionStrategyPrototype) {
+    this.watchdog = watchdog;
+    this.innerCallable = innerCallable;
+    this.executor = executor;
+    this.resumptionStrategyPrototype = resumptionStrategyPrototype;
+  }
+
+  @Override
+  public void call(
```

Comment:
The `call()` returning `void` is still disturbing to me =(

Reply:
Nope sure how to address this. I think this is similar to void api result algorithm. 

As a sidenote, the UnaryCallable version flat out ignores it:
https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/RetryingCallable.java#L63

Reply:
Oops, I misunderstood the comment. The `call()` is part of the ServerStreamingCallable interface. Changing it is out of scope for this PR. Please open a github issue to discuss api changes to the server streaming callable interface